plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -119,6 +122,7 @@ newtype ExCPU = ExCPU CostingInteger
   deriving (Semigroup, Monoid) via (Sum CostingInteger)
   deriving (FromJSON, ToJSON) via CostingInteger
   deriving Serialise via CostingInteger
+  deriving anyclass NoThunks
```

Comment:
why not newtype deriving? I guess that will give us a weird implementation of `showTypeOf`?

Reply:
Yeah, if we use newtype deriving, then `showTypeOf` won't tell us if its `ExMemory` or `ExCPU` since they are both `CostingInteger`.